src/transformer.ts

```diff
@@ -0,0 +1,222 @@
+  // beforeTsTransformers.push(createTransformer(host, typeChecker, (sourceFile, sourceMapper) => {
+  //   sourceMapper.addMapping(sourceFile, {position: 0, line: 0, column: 0}, {position: 0, line: 0,
+  //   column: 0}, sourceFile.text.length); return sourceFile.text;
+  // }));
+  // add user supplied transformers
+  const afterTsTransformers: Array<ts.TransformerFactory<ts.SourceFile>> = [];
+  if (customTransformers) {
+    if (customTransformers.beforeTsickle) {
+      beforeTsTransformers.unshift(...customTransformers.beforeTsickle);
+    }
+
+    if (customTransformers.beforeTs) {
+      beforeTsTransformers.push(...customTransformers.beforeTs);
+    }
+    if (customTransformers.afterTs) {
+      afterTsTransformers.push(...customTransformers.afterTs);
+    }
+  }
+  customTransformers = createCustomTransformers({
+    before: beforeTsTransformers.map(tf => skipTransformForSourceFileIfNeeded(host, tf)),
+    after: afterTsTransformers.map(tf => skipTransformForSourceFileIfNeeded(host, tf))
+  });
+
+  const writeFileDelegate = writeFile || tsHost.writeFile.bind(tsHost);
+  const modulesManifest = new ModulesManifest();
+  const writeFileImpl =
+      (fileName: string, content: string, writeByteOrderMark: boolean,
+       onError?: (message: string) => void, sourceFiles?: ts.SourceFile[]) => {
+        if (path.extname(fileName) !== '.map') {
+          if (tsOptions.inlineSourceMap) {
+            content = combineInlineSourceMaps(program, fileName, content);
+          } else {
+            content = removeInlineSourceMap(content);
+          }
+          content = es5processor.convertCommonJsToGoogModuleIfNeeded(
+              host, options, modulesManifest, fileName, content);
+        } else {
+          content = combineSourceMaps(program, fileName, content);
+        }
+        writeFileDelegate(fileName, content, writeByteOrderMark, onError, sourceFiles);
+      };
+
+  const {diagnostics: tsDiagnostics, emitSkipped, emittedFiles} = program.emit(
+      targetSourceFile, writeFileImpl, cancellationToken, emitOnlyDtsFiles, customTransformers);
+
+  const externs: {[fileName: string]: string} = {};
+  if (options.transformTypesToClosure) {
+    const sourceFiles = targetSourceFile ? [targetSourceFile] : program.getSourceFiles();
+    sourceFiles.forEach(sf => {
+      if (tsickle.isDtsFileName(sf.fileName) && host.shouldSkipTsickleProcessing(sf.fileName)) {
+        return;
+      }
+      const {output, diagnostics} = tsickle.writeExterns(typeChecker, sf, host, options);
+      if (output) {
+        externs[sf.fileName] = output;
+      }
+      if (diagnostics) {
+        tsickleDiagnostics.push(...diagnostics);
+      }
+    });
+  }
+  // All diagnostics (including warnings) are treated as errors.
+  // If we've decided to ignore them, just discard them.
+  // Warnings include stuff like "don't use @type in your jsdoc"; tsickle
+  // warns and then fixes up the code to be Closure-compatible anyway.
```

Comment:
The rule here is that we error on google3 code, and don't error on third-party code.  Is that still preserved here?